src/app/frontend/replicasetlist/replicasetlist_state.js

```diff
@@ -32,15 +33,32 @@ export default function stateConfig($stateProvider) {
       replicaSets: resolveReplicaSets,
     },
     templateUrl: 'replicasetlist/replicasetlist.html',
+    onEnter: initState,
   });
 }
 
 /**
+ * Avoids entering replica set list page when there are no replica sets.
+ * Used f.e. when last replica set gets deleted.
+ * Transition to: zerostate
+ * @param {!ui.router.$state} $state
+ * @param {!angular.$timeout} $timeout
+ * @param {!backendApi.ReplicaSetList} replicaSets
+ * @ngInject
+ */
+function initState($state, $timeout, replicaSets) {
+  if (replicaSets.replicaSets.length === 0) {
+    // allow original state change to finish before redirecting to new state to avoid error
+    $timeout(() => {$state.go(zerostate);});
+  }
+}
+
+/**
  * @param {!angular.$resource} $resource
  * @return {!angular.$q.Promise}
  * @ngInject
  */
-export function resolveReplicaSets($resource) {
```

Comment:
Definitely. Maybe a quick fix in other files? I see this pattern applied everywhere else.